mr/joiner_executor.cc

```diff
@@ -64,7 +50,7 @@ void JoinerExecutor::Run(const std::vector<const InputBase*>& inputs, detail::Ta
   pool_->AwaitOnAll([&](unsigned index, IoContext&) {
     per_io_.reset(new PerIoStruct(index));
     per_io_->raw_context.reset(runner_->CreateContext());
-    per_io_->process_fd = fibers::fiber{&JoinerExecutor::ProcessInputQ, this, tb};
+    per_io_->process_fd.emplace_back(fibers::fiber{&JoinerExecutor::ProcessInputQ, this, tb});
```

Comment:
nit: you might omit `fibers::fiber` and just pass the arguments directly. vector should be able to create the object in-place.